datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1388,38 +1491,50 @@ impl SMJStream {
                         };
 
                     // Push the filtered batch which contains rows passing join filter to the output
-                    let filtered_batch =
-                        compute::filter_record_batch(&output_batch, mask)?;
-                    self.output_record_batches.push(filtered_batch);
+                    if self.join_type == JoinType::Left {
+                        self.output_record_batches
+                            .batches
+                            .push(output_batch.clone());
```

Comment:
For Left WIP we pass the original non filtered batch because filtering will be done later.